netty/src/main/java/io/grpc/netty/HandlerSettings.java

```diff
@@ -41,36 +41,47 @@
 @VisibleForTesting
 public final class HandlerSettings {
 
+  private static volatile boolean enabled;
+
   private static boolean autoFlowControlOn;
   // These will be the most recently created handlers created using NettyClientTransport and
   // NettyServerTransport
   private static AbstractNettyHandler clientHandler;
   private static AbstractNettyHandler serverHandler;
 
   static void setAutoWindow(AbstractNettyHandler handler) {
-    handler.setAutoTuneFlowControl(autoFlowControlOn);
-    if (handler instanceof NettyClientHandler) {
-      clientHandler = handler;
-    } else if (handler instanceof NettyServerHandler) {
-      serverHandler = handler;
-    } else {
-      throw new RuntimeException("Expecting NettyClientHandler or NettyServerHandler");
+    if (!enabled) {
+      return;
+    }
+    synchronized (HandlerSettings.class) {
+      handler.setAutoTuneFlowControl(autoFlowControlOn);
+      if (handler instanceof NettyClientHandler) {
+        clientHandler = handler;
+      } else if (handler instanceof NettyServerHandler) {
+        serverHandler = handler;
+      } else {
+        throw new RuntimeException("Expecting NettyClientHandler or NettyServerHandler");
+      }
     }
   }
 
-  public static void autoWindowOn(boolean autoFlowControl) {
+  public static void enable(boolean enable) {
```

Comment:
To make it a noop when not running in a test.  It also makes it skip synchronization.